Analysis/Tasks/PWGHF/HFCandidateCreatorXicc.cxx

```diff
@@ -0,0 +1,271 @@
+        trackParVarPi.propagateToDCA(primaryVertex, magneticField, &impactParameter1);
+
+        // get uncertainty of the decay length
+        double phi, theta;
+        getPointDirection(array{collision.posX(), collision.posY(), collision.posZ()}, secondaryVertexXicc, phi, theta);
+        auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixPCA, phi, theta));
+        auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixPCA, phi, 0.));
+
+        int hfFlag = 1 << DecayType::XiccToXicPi;
+
+        rowCandidateBase(collision.globalIndex(),
+                         collision.posX(), collision.posY(), collision.posZ(),
+                         secondaryVertexXicc[0], secondaryVertexXicc[1], secondaryVertexXicc[2],
+                         errorDecayLength, errorDecayLengthXY,
+                         chi2PCA,
+                         pvecxic[0], pvecxic[1], pvecxic[2],
+                         pvecpion[0], pvecpion[1], pvecpion[2],
+                         impactParameter0.getY(), impactParameter1.getY(),
+                         std::sqrt(impactParameter0.getSigmaY2()), std::sqrt(impactParameter1.getSigmaY2()),
+                         xicCand.globalIndex(), trackpion.globalIndex(),
+                         hfFlag);
+      } // if on selected Xicc
+    }   // loop over candidates
+  }     // end of process
+};      //end of struct
+
+/// Extends the base table with expression columns.
+struct HfCandidateCreatorXiccExpressions {
+  Spawns<aod::HfCandXiccExt> rowCandidateXicc;
+  void init(InitContext const&) {}
+};
+
+/// Performs MC matching.
+struct HfCandidateCreatorXiccMc {
+  Produces<aod::HfCandXiccMCRec> rowMCMatchRec;
+  Produces<aod::HfCandXiccMCGen> rowMCMatchGen;
+
+  void process(aod::HfCandXicc const& candidates,
+               aod::HfCandProng3,
+               aod::BigTracksMC const& tracks,
+               aod::McParticles const& particlesMC)
+  {
+    int indexRec = -1;
+    int8_t sign = 0;
+    int8_t flag = 0;
+    int8_t origin = 0;
+
+    // Match reconstructed candidates.
+    for (auto& candidate : candidates) {
+      //Printf("New rec. candidate");
+      flag = 0;
+      origin = 0;
+
+      auto xicCand = candidate.index0();
+      auto arrayDaughters = array{xicCand.index0_as<aod::BigTracksMC>(),
+                                  xicCand.index1_as<aod::BigTracksMC>(),
+                                  xicCand.index2_as<aod::BigTracksMC>(),
+                                  candidate.index1_as<aod::BigTracksMC>()};
+
+      // Ξcc±± → p± K∓ π± π±
+      //Printf("Checking Ξcc±± → p± K∓ π± π±");
+      indexRec = RecoDecay::getMatchedMCRec(particlesMC, arrayDaughters, pdg::Code::kXiCCPlusPlus, array{+kProton, -kKPlus, +kPiPlus, +kPiPlus}, true, &sign, 3);
+      if (indexRec > -1) {
+        flag = 1 << DecayType::XiccToXicPi;
+      }
```

Comment:
Shouldn't depth 2 be used?
Don't you need to also check Ξcc -> Ξc?

Reply:
Yes, I am testing a version.